service/raft/group0_state_machine.cc

```diff
@@ -121,7 +121,14 @@ future<> group0_state_machine::apply(std::vector<raft::command_cref> command) {
         void add(group0_command&& cmd, size_t added_size) {
             slogger.trace("add to merging set new_state_id: {}", cmd.new_state_id);
             auto m = convert_history_mutation(std::move(cmd.history_append), sm._sp.data_dictionary());
-            last_group0_state_id = std::max(last_group0_state_id, cmd.new_state_id);
+            // Set `last_group0_state_id` to the maximum of the current value and `cmd.new_state_id`,
+            // but make sure we compare them the same way timeuuids are compared in clustering keys
+            // (i.e. in the same order that the history table is sorted).
+            auto cmp = timeuuid_type->compare(
+                timeuuid_type->decompose(last_group0_state_id), timeuuid_type->decompose(cmd.new_state_id));
```

Comment:
I understand the urgency of the fix, but adding two decompose and then a virtual call for comparing two integers is too much. Could you please implement UUID::timeuuid_tricompare following the impplementation of timeuuid_tri_compare function?